projects/components/src/link/link.component.ts

```diff
@@ -1,45 +1,41 @@
-import { ChangeDetectionStrategy, Component, Input } from '@angular/core';
-import { NavigationService } from '@hypertrace/common';
+import { ChangeDetectionStrategy, Component, Input, OnChanges } from '@angular/core';
+import { NavigationExtras } from '@angular/router';
+import { NavigationPath, NavigationService } from '@hypertrace/common';
 import { isEmpty } from 'lodash-es';
 
 @Component({
   selector: 'ht-link',
   styleUrls: ['./link.component.scss'],
   changeDetection: ChangeDetectionStrategy.OnPush,
   template: `
-    <a *ngIf="this.shouldShowLink" class="ht-link" (click)="this.navigateToUrl()">
+    <a
+      *ngIf="this.navigationParams"
+      class="ht-link"
+      [routerLink]="this.navigationParams[0]"
+      [queryParams]="this.navigationParams[1].queryParams"
```

Comment:
how do we want to handle query params?

Currently, the only input the component takes is the URL. if we expect the URL to be inclusive of query param string, then we have to extract them for this to function. Else, we can just use the URL string and this is redundant.

The downstream method `buildInternalNavigationParams` consumes queryparams only from navigationExtras argument which we are not supplying in L38.

Reply:
Given my proposal above, this input could accept a `string | NavigationParams` since that's what the navigate api would accept.

Reply:
When I tested this, url had the path and query params. See an example here. https://github.com/Traceableai/traceable-ui/pull/1374/files#r498031200

So, lets just have two inputs in this component, path: NavigationPath (which can be string or array) and params: NavigationParams.

Reply:
the params as I described above would already self contain both the path and query params